app/frontend/models/requirement-block.ts

```diff
@@ -39,14 +39,9 @@ export const RequirementBlockModel = types
 
       if (response.ok) {
         applySnapshot(self, response.data.data)
-
-        // Get latest data for current page, sort and filters
-        yield self.rootStore.requirementBlockStore.fetchRequirementBlocks()
```

Comment:
this was causing a "recursively references self as base type" in requirement block store.
The re-fetch is now called outside of the update

Reply:
I don't see where the fetch is being called, but I am approving

Reply:
app/frontend/components/domains/requirements-library/requirements-block-modal/index.tsx:83